libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -153,13 +149,19 @@ uint8_t AP_ESC_Telem::get_num_active_escs() const {
 // return the whether all the motors in servo_channel_mask are running
 bool AP_ESC_Telem::are_motors_running(uint32_t servo_channel_mask, float min_rpm, float max_rpm) const
 {
-    const uint32_t now = AP_HAL::micros();
 
     for (uint8_t i = 0; i < ESC_TELEM_MAX_ESCS; i++) {
         if (BIT_IS_SET(servo_channel_mask, i)) {
             const volatile AP_ESC_Telem_Backend::RpmData& rpmdata = _rpm_data[i];
-            // we choose a relatively strict measure of health so that failsafe actions can rely on the results
-            if (!rpm_data_within_timeout(rpmdata, now, ESC_RPM_CHECK_TIMEOUT_US)) {
+            // copy the last_updated_us timestamp to avoid any race issues
+            const uint32_t last_updated_us = rpmdata.last_update_us;
+            const uint32_t now = AP_HAL::micros();
+            // we choose a stricter timeout than the data_valid flag so that failsafe actions can rely on the results
+            if (now - last_updated_us > ESC_RPM_CHECK_TIMEOUT_US) {
```

Comment:
Please use AP_HAL::timeout_expired() here

Reply:
Surely this is just rpm_data_within_timeout() in which case better to leave it in.